Sources/Filters/General/ImageMarchingSquares/index.js

```diff
@@ -129,53 +157,70 @@ function vtkImageMarchingSquares(publicAPI, model) {
       return;
     }
 
+    if (
+      model.slicingMode == null ||
+      model.slicingMode < 0 ||
+      model.slicingMode > 2
+    ) {
+      vtkErrorMacro('Invalid or missing slicing mode');
+      return;
+    }
+
     console.time('msquares');
 
     // Retrieve output and volume data
     const origin = input.getOrigin();
     const spacing = input.getSpacing();
     const dims = input.getDimensions();
-    const s = input.getPointData().getScalars().getData();
+    const extent = input.getExtent();
+    const increments = input.computeIncrements(extent);
+    const scalars = input.getPointData().getScalars().getData();
+    const [kernelX, kernelY] = getKernels();
 
     // Points - dynamic array
-    const pBuffer = [];
+    const points = [];
 
     // Cells - dynamic array
-    const lBuffer = [];
+    const lines = [];
 
     // Ensure slice is valid
-    const slice = dims[0] * dims[1];
     let k = Math.round(model.slice);
-    if (k >= dims[2]) {
+    if (k >= dims[model.slicingMode]) {
       k = 0;
     }
 
     // Loop over all contour values, and then pixels, determine case and process
     for (let cv = 0; cv < model.contourValues.length; ++cv) {
-      for (let j = 0; j < dims[1] - 1; ++j) {
-        for (let i = 0; i < dims[0] - 1; ++i) {
+      for (let j = 0; j < dims[kernelY] - 1; ++j) {
+        for (let i = 0; i < dims[kernelX] - 1; ++i) {
+          const ijk = [0, 0, 0];
```

Comment:
you can instantiate `ijk` before the `for` loops and set each element within each `for` loop

Reply:
Good point, just added that to the PR.

Reply:
Ah yes, should be good now.

Reply:
(feel free to squash your commits in a unique commit (`git rebase -i HEAD~6`)